test/Dialect/Verif/basic.mlir

```diff
@@ -35,6 +35,27 @@ verif.cover %true label "foo3" : i1
 verif.cover %s : !ltl.sequence
 verif.cover %p : !ltl.property
 
+//===----------------------------------------------------------------------===//
+// Formal
+//===----------------------------------------------------------------------===//
+hw.module @Foo(in %0 "0": i1, in %clk "clk": !seq.clock, out "" : i1, out "1" : i1) {
+  %1 = seq.from_clock %clk
+  hw.output %0 , %1: i1, i1
+ }
+
+// CHECK: verif.formal sym @formal1(k : 20 : i64 type : bmc) {
+verif.formal sym @formal1(k : 20 type : bmc) {
```

Comment:
The assembly format in the parentheses is a bit weird. Maybe something like `(k = 20, type = bmc)` would be easier to understand.

Reply:
yeah I'm just not very good at writing the assembly regex format and couldn't figure out how to make the commas only needed if the two arguments were given. Any idea?

Reply:
Why are they optional? If they have a default value, you can make them default-valued and just print them in the assembly format regardless.